docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,121 @@
+On the other hand, all K8s clusters in production deployments use CNIs which support NP.
+
+## Implementation
+
+The `from`/`to` element of `ingress`/`egress`  
+[NetworkPolicySpec](https://kubernetes.io/docs/reference/kubernetes-api/policy-resources/network-policy-v1/#NetworkPolicySpec) 
+allows to restrict incoming/outgoing network traffic based on `podSelectors` (labels based), and/or 
+`namespaceSelectors` (labels based) and/or `ipBlocks`. The last parameter can be used to restrict cross cluster connections 
+
+We suggest, after deployment a data plane module, to create an instance of a network policy that will restrict the 
+module connectivity.
+
+### Backward compatibility and Network Policies isolation
+
+Addition of NP isolation can change the existing Fybrik behavior. In order to support backward compatibility, we suggest
+to add NP as an optional feature. A new entry `global.isNPEnabled` will be added into the Fybrik chart `values.yaml`.
+NP will be evoked only if this value is set to `true`.
+
+### FybrikApplication extensions
+
+`FybrikApplicationSpec` has the “[selector](https://fybrik.io/v1.2/reference/crds/#fybrikapplicationspecselector)” 
+element, which is a combination of `clusterName` and `workloadSelector`. Unfortunately, based on this information we can 
+implement only one label-based `podSelector`, which will allow incoming connections from pods in the namespace where the NP 
+instance is deployed and have the defined label. In order to restrict incoming connections to Fybrik modules, the NP
+instances should be deployed in the same namespace where the modules are running. However, Fybrik separates user's 
+objects, e.g. workloads and Fybrik objects, e.g. modules. Therefore, `FybrikApplicationSpec.selector` should be extended.
+
+#### Suggestions
+
+Extend the `FybrikApplicationSpec.selector` element with an optional array of namespaces names, and an optional blocks
+of IPs. These extensions will not break CRDs backward compatability.
+
+The `Selector` will have the following structure: 
+
+```go
+type Selector struct {
+	// Cluster name 
+	//+optional 
+	ClusterName string `json:"clusterName"`
+	
+	WorkloadSelector enables to connect the resource to the application 
+	// Applications labels should match the labels in the selector. 
+	//+required 
+	WorkloadSelector metav1.LabelSelector `json:"workloadSelector"`
+	
+	// Namespaces where user application might run
+	// +optional 
+	Namespaces []string `json:"namespaces"`
+
+	// IPBlocks define policy on particular IPBlocks.
+	// the structure of the IPBlock is defined at https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#ipblock-v1-networking-k8s-io
+	// +optional
+	IPBlocks []*IPBlock `json:"ipBlocks,omitempty"`
+}
+```
+The mapping the `Selector` elements into the `from` NP ingress definitions will be done according the following algorithm:
+- if `Namespaces` is not empty, for each entry in the array will be created a 
```

Comment:
If Namespaces is not empty but WorkloadSelector is empty, then it means any pods in the namespaces, correct?

Reply:
correct, added the explanation.